lib/exchanges/bitfinex.js

```diff
@@ -44,28 +31,25 @@ function openWebSocket(pair) {
     "event": "subscribe",
     "channel": "book",
     "pair": pair,
-    "prec": "R0",
+    "prec": "P0",
     "len":"25"
   });
   ws.on('open', () => {
     ws.send(msg);
   });
 
-  var counter = 3;
+  var flag = true;
 
-  // TODO: Can we use the flags parameter instead of this counting system?
   ws.on('message', function(data, flags) {
+
     data = JSON.parse(data);
 
     // http://docs.bitfinex.com/#websocket
 
-    if (counter > 1) {
-      counter--;
-    } else if (counter == 1) {
-      // Bitfinex 3rd message is a snapshot of the order book
+    if (!data.event && flag) {
       parseSnapshot(data);
-      counter--;
-    } else if (counter == 0) {
+      flag = false;
+    } else if (!flag) {
```

Comment:
Adding explaining comments in my next commit, essentially I changed it to not need a counter system based on the contents of the messages.